Fail finding libelf & zlib although exists

I'm trying to compile the repo using the Makefile but I get the following error:

```bash
➜  tracee git:(main) ✗ make        
make ./dist/btfhub
GOOS=linux CC=clang GOARCH=amd64 CGO_CFLAGS="-I/home/yuvalgold/workspace/aqua/tracee/dist/libbpf" CGO_LDFLAGS="-lelf -lz /home/yuvalgold/workspace/aqua/tracee/dist/libbpf/libbpf.a" go build \
	-tags core,ebpf \
	-ldflags="-w \
		-extldflags \"\" \
		-X main.version=\"v0.6.5-201-gbc05bc0\" \
		" \
	-v -o dist/tracee-ebpf \
	./cmd/tracee-ebpf
make[1]: 'dist/btfhub' is up to date.
runtime/cgo
# runtime/cgo
/usr/bin/ld: cannot find -lelf
/usr/bin/ld: cannot find -lz
clang-13: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:468: dist/tracee-ebpf] Error 2
```

It seems that the validations within the Makefile pass

```bash
➜  tracee git:(main) ✗ pkg-config --libs libelf
-L/home/linuxbrew/.linuxbrew/Cellar/elfutils/0.186/lib -lelf
➜  tracee git:(main) ✗ pkg-config --libs zlib  
-L/home/linuxbrew/.linuxbrew/Cellar/zlib/1.2.11/lib -lz
```

But somehow it doesn't pass to `clang`. Any idea?

<details>
<summary>Environment (click to expand)</summary>

```bash
➜  tracee git:(main) ✗ ld --version
GNU ld (GNU Binutils for Ubuntu) 2.37
Copyright (C) 2021 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or (at your option) a later version.
This program has absolutely no warranty.

➜  tracee git:(main) ✗ cat /etc/os-release 
PRETTY_NAME="Ubuntu 21.10"
NAME="Ubuntu"
VERSION_ID="21.10"
VERSION="21.10 (Impish Indri)"
VERSION_CODENAME=impish
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=impish

➜  tracee git:(main) ✗ make env
---------------------------------------
Makefile Environment:
---------------------------------------
PARALLEL                 8
---------------------------------------
CLANG_VERSION            13
GO_VERSION               1.17
---------------------------------------
CMD_CLANG                clang
CMD_GIT                  git
CMD_GO                   go
CMD_INSTALL              install
CMD_LLC                  llc
CMD_MD5                  md5sum
CMD_OPA                  opa 
CMD_PKGCONFIG            pkg-config
CMD_STRIP                llvm-strip
---------------------------------------
LIB_ELF                  libelf
LIB_ZLIB                 zlib
---------------------------------------
VERSION                  v0.6.5-201-gbc05bc0
LAST_GIT_TAG             v0.6.5-201-gbc05bc0
BPF_NOCORE_TAG           5_10_0-1050-oem.v0_6_5-201-gbc05bc0
---------------------------------------
UNAME_M                  x86_64
UNAME_R                  5.10.0-1050-oem
ARCH                     x86_64
LINUX_ARCH               x86
---------------------------------------
OUTPUT_DIR               ./dist
---------------------------------------
KERN_RELEASE             5.10.0-1050-oem
KERN_BUILD_PATH          /lib/modules/5.10.0-1050-oem/build
KERN_SRC_PATH            /lib/modules/5.10.0-1050-oem/build
---------------------------------------
LIBBPF_CFLAGS            
LIBBPF_LDLAGS            
LIBBPF_SRC               ./3rdparty/libbpf/src
---------------------------------------
STATIC                   0
---------------------------------------
BPF_VCPU                 v2
TRACEE_EBPF_OBJ_SRC      ./pkg/ebpf/c/tracee.bpf.c
---------------------------------------
GO_ARCH                  amd64
GO_TAGS_EBPF             core,ebpf
GO_TAGS_RULES            
---------------------------------------
CUSTOM_CGO_CFLAGS        -I/home/yuvalgold/workspace/aqua/tracee/dist/libbpf
CUSTOM_CGO_LDFLAGS       -lelf -lz /home/yuvalgold/workspace/aqua/tracee/dist/libbpf/libbpf.a
CGO_EXT_LDFLAGS_EBPF     
CGO_EXT_LDFLAGS_RULES    
---------------------------------------
GO_ENV_EBPF              GOOS=linux CC=clang GOARCH=amd64 CGO_CFLAGS=-I/home/yuvalgold/workspace/aqua/tracee/dist/libbpf CGO_LDFLAGS=-lelf -lz /home/yuvalgold/workspace/aqua/tracee/dist/libbpf/libbpf.a
GO_ENV_RULES             GOOS=linux CC=clang GOARCH=amd64 CGO_CFLAGS= CGO_LDFLAGS=
---------------------------------------
TRACEE_EBPF_SRC          ./cmd/tracee-ebpf/main.go ./cmd/tracee-ebpf/internal/flags/flags-capture.go ./cmd/tracee-ebpf/internal/flags/flags-filter.go ./cmd/tracee-ebpf/internal/flags/flags-output.go ./cmd/tracee-ebpf/printer.go ./pkg/ebpf/events_processor.go ./pkg/ebpf/write_capture.go ./pkg/ebpf/events_derived.go ./pkg/ebpf/net_events.go ./pkg/ebpf/tracee.go ./pkg/ebpf/events_pipeline.go ./pkg/ebpf/events_amd64.go ./pkg/ebpf/events_arm64.go ./pkg/ebpf/user_mode_events.go ./pkg/ebpf/filters.go ./pkg/ebpf/initialization/kconfig.go ./pkg/ebpf/events_definitions.go ./pkg/ebpf/argprinters.go ./pkg/bufferdecoder/protocol.go ./pkg/bufferdecoder/decoder.go ./pkg/bufferdecoder/eventsreader.go ./pkg/bufferdecoder/net_decoder.go ./pkg/counter/counter.go ./pkg/utils/environment/cpu_analyzing.go ./pkg/containers/containers.go ./pkg/ebpf/events_processor.go ./pkg/ebpf/write_capture.go ./pkg/ebpf/events_derived.go ./pkg/ebpf/net_events.go ./pkg/ebpf/tracee.go ./pkg/ebpf/events_pipeline.go ./pkg/ebpf/events_amd64.go ./pkg/ebpf/events_arm64.go ./pkg/ebpf/user_mode_events.go ./pkg/ebpf/filters.go ./pkg/ebpf/initialization/kconfig.go ./pkg/ebpf/events_definitions.go ./pkg/ebpf/argprinters.go ./pkg/test/gob/gob.go ./pkg/rules/engine/engine.go ./pkg/rules/benchmark/signature/wasm/signatures.go ./pkg/rules/benchmark/signature/rego/signatures.go ./pkg/rules/benchmark/signature/golang/anti_debugging.go ./pkg/rules/benchmark/signature/golang/noop.go ./pkg/rules/benchmark/signature/golang/code_injection.go ./pkg/rules/regosig/traceerego.go ./pkg/rules/regosig/aio.go ./pkg/rules/regosig/mapper.go ./pkg/rules/metrics/stats.go ./pkg/rules/metrics/prometheus.go ./pkg/procinfo/procinfo.go ./pkg/metrics/stats.go ./pkg/metrics/prometheus.go ./pkg/events/sorting/pool.go ./pkg/events/sorting/sorting.go ./pkg/events/sorting/cpu_queue.go ./pkg/events/sorting/queue.go ./pkg/capabilities/capabilities.go ./pkg/bucketscache/bucketscache.go
TRACEE_EBPF_SRC_DIRS     ./cmd/tracee-ebpf/ ./pkg/ebpf ./pkg
---------------------------------------
TRACEE_RULES_SRC         ./cmd/tracee-rules/output.go ./cmd/tracee-rules/main_test.go ./cmd/tracee-rules/main.go ./cmd/tracee-rules/signature_test.go ./cmd/tracee-rules/input.go ./cmd/tracee-rules/input_test.go ./cmd/tracee-rules/output_test.go ./cmd/tracee-rules/signature.go ./pkg/rules/engine/engine_test.go ./pkg/rules/engine/engine.go ./pkg/rules/benchmark/benchmark_test.go ./pkg/rules/benchmark/signature/wasm/signatures.go ./pkg/rules/benchmark/signature/rego/signatures.go ./pkg/rules/benchmark/signature/golang/anti_debugging.go ./pkg/rules/benchmark/signature/golang/noop.go ./pkg/rules/benchmark/signature/golang/code_injection.go ./pkg/rules/benchmark/helpers_test.go ./pkg/rules/regosig/traceerego.go ./pkg/rules/regosig/common_test.go ./pkg/rules/regosig/aio.go ./pkg/rules/regosig/aio_test.go ./pkg/rules/regosig/traceerego_test.go ./pkg/rules/regosig/mapper.go ./pkg/rules/regosig/mapper_test.go ./pkg/rules/metrics/stats.go ./pkg/rules/metrics/prometheus.go
TRACEE_RULES_SRC_DIRS    ./cmd/tracee-rules/ ./pkg/rules/
---------------------------------------
GOSIGNATURES_DIR         signatures/golang
GOSIGNATURES_SRC         signatures/golang/stdio_over_socket.go signatures/golang/kubernetes_api_connection.go signatures/golang/export.go
---------------------------------------
REGO_SIGNATURES_DIR      signatures/rego
REGO_SIGNATURES_SRC      signatures/rego/k8s_service_account_token.rego signatures/rego/dropped_executable.rego signatures/rego/ld_preload.rego signatures/rego/anti_debugging_ptraceme.rego signatures/rego/fileless_execution.rego signatures/rego/helpers.rego signatures/rego/cgroup_release_agent_modification.rego signatures/rego/dynamic_code_loading.rego signatures/rego/disk_mount.rego signatures/rego/illegitimate_shell.rego signatures/rego/code_injection.rego signatures/rego/kernel_module_loading.rego signatures/rego/kubernetes_certificate_theft_attempt.rego
---------------------------------------
```

</details>

--- a/Makefile
+++ b/Makefile
@@ -52,6 +52,10 @@ CMD_OPA ?= opa # https://github.com/open-policy-agent/opa/releases/download/v0.3
 LIB_ELF ?= libelf
 LIB_ZLIB ?= zlib
 
+define pkg_config
+	$(CMD_PKGCONFIG) --libs $(1)
+endef
+
 .checklib_%: \
 	| .check_$(CMD_PKGCONFIG)
 #
@@ -442,7 +446,7 @@ ifeq ($(STATIC), 1)
 endif
 
 CUSTOM_CGO_CFLAGS = "-I$(abspath $(OUTPUT_DIR)/libbpf)"
-CUSTOM_CGO_LDFLAGS = "-lelf -lz $(abspath $(OUTPUT_DIR)/libbpf/libbpf.a)"
+CUSTOM_CGO_LDFLAGS = "$(shell $(call pkg_config, $(LIB_ELF))) $(shell $(call pkg_config, $(LIB_ZLIB))) $(abspath $(OUTPUT_DIR)/libbpf/libbpf.a)"
 
 GO_ENV_EBPF =
 GO_ENV_EBPF += GOOS=linux

